samples/download-preplanned-map-area/src/main/java/com/esri/arcgismaps/sample/downloadpreplannedmaparea/screens/DownloadPreplannedMapAreaScreen.kt

```diff
@@ -0,0 +1,197 @@
+                    onClick = { showBottomSheet = true }) {
+                    Icon(
+                        Icons.Filled.Settings, contentDescription = "Select map"
+                    )
+                }
+            }
+        })
+}
+
+@Composable
+fun DownloadPreplannedMap(
+    showOnlineMap: () -> Unit,
+    downloadOrShowMap: (PreplannedMapAreaInfo) -> Unit,
+    preplannedMapAreaInfoFlow: StateFlow<List<PreplannedMapAreaInfo>>
+) {
+    Column(
+        modifier = Modifier
+            .wrapContentSize()
+            .background(MaterialTheme.colorScheme.background)
+            .padding(12.dp)
+            .verticalScroll(rememberScrollState()),
+        horizontalAlignment = Alignment.Start,
+    ) {
+        Text(
+            modifier = Modifier
+                .align(Alignment.CenterHorizontally)
+                .padding(16.dp),
+            text = "Select map",
+            style = MaterialTheme.typography.titleLarge
+        )
+        Card(modifier = Modifier.wrapContentSize()) {
+            Row(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(8.dp)
+                    .clickable { showOnlineMap() },
+                horizontalArrangement = Arrangement.spacedBy(4.dp),
+                verticalAlignment = CenterVertically,
+            ) {
+                Text("Web map (online)")
+            }
+        }
+        Text(text = "Preplanned map areas", modifier = Modifier.padding(top = 16.dp, start = 8.dp, bottom = 4.dp))
+        Card(modifier = Modifier.wrapContentSize()) {
+            preplannedMapAreaInfoFlow.collectAsState().value.forEach {
+                Row(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(8.dp)
+                        .clickable {
+                            downloadOrShowMap(it)
+                        },
+                    horizontalArrangement = Arrangement.spacedBy(4.dp),
+                    verticalAlignment = CenterVertically,
+                ) {
+                    if (!it.isDownloaded) {
+                        if (it.progress <= 0) {
+                            Icon(
+                                painter = painterResource(R.drawable.baseline_download_for_offline_24),
+                                contentDescription = "Download"
+                            )
+                        } else {
+                            CircularProgressIndicator(
+                                modifier = Modifier.requiredSize(18.dp),
+                                progress = { it.progress })
+                        }
+                    }
+                    Text(
+                        text = it.name, color = if (it.isDownloaded) {
+                            Color.Black
+                        } else {
+                            Color.Gray
+                        }
```

Comment:
Hardcoded colors are used here, would suggest using some theme colors: 

```suggestion
                        text = it.name, color = if (it.isDownloaded) {
                            MaterialTheme.colorScheme.onSurface
                        } else {
                            MaterialTheme.colorScheme.outline
                        }
```